public/js/components/Editor.js

```diff
@@ -71,45 +71,53 @@ const Editor = React.createClass({
     }
   },
 
-  clearDebugLine(line) {
-    this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
-  },
-
-  setDebugLine(line) {
-    this.editor.codeMirror.addLineClass(line - 1, "line", "debug-line");
-    this.editor.alignLine(line);
-  },
-
-  setSourceText(newSourceText, oldSourceText) {
-    if (newSourceText.get("loading")) {
-      this.setText("Loading...");
-      return;
+  updateDebugLine(prevProps, nextProps) {
+    if (prevProps.selectedFrame) {
+      const line = prevProps.selectedFrame.location.line;
+      this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
     }
-
-    if (newSourceText.get("error")) {
-      this.setText("Error");
-      console.error(newSourceText.get("error"));
-      return;
+    if (nextProps.selectedFrame) {
+      const line = nextProps.selectedFrame.location.line;
```

Comment:
I'm not quite convinced; I'd like to wait and maybe do it later. I think it's good for the components to use terms that are more concrete like `selectedFrame` generally instead of receiving things like `debugLine` from somewhere else. I think those terms are a little vague, with highlighted/debug/etc being sort of similar. We can see how it feels over time.